### PR TITLE
CAMEL-20044 Removing the lastResult used in breakOnFirstError 

### PR DESCRIPTION
**Pull request by CodeSmell**
# Description

This PR is adding onto what is in https://github.com/apache/camel/pull/11959
It shows that we can remove the `lastResult` from the code if we go with the changes proposed in this PR

# Target

camel-3.21.x

# Tracking
CAMEL-20044

### CODE REVIEW

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/consumer/support/KafkaRecordProcessorFacade.java [hunk 1]
```diff
@@ -54,49 +54,53 @@ private boolean isStopping() {
...
+                    if (!consumerListener.afterProcess(result)) {
                         commitManager.commit(partition);
-                        return lastResult;
+                        return result;
                     }
                 }
             }
 
-            if (!lastResult.isBreakOnErrorHit()) {
+            if (!result.isBreakOnErrorHit()) {
```
Comment (CodeSmell) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/consumer/support/KafkaRecordProcessorFacade.java:96]: this part of the code is the only reason we need the offset  in `ProcessedResult`
may be worth trying to move it inside `createExchange`

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/consumer/support/KafkaRecordProcessorFacade.java [hunk 2]
```diff
@@ -126,7 +129,7 @@ private ProcessingResult processRecord(
 
         ProcessingResult currentResult
                 = kafkaRecordProcessor.processExchange(exchange, partition, partitionHasNext,
-                        recordHasNext, record, lastResult, camelKafkaConsumer.getExceptionHandler());
+                        recordHasNext, record, camelKafkaConsumer.getExceptionHandler());
 
         if (!currentResult.isBreakOnErrorHit()) {
             commitManager.recordOffset(partition, currentResult.getPartitionLastOffset());
```
Comment (CodeSmell) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/consumer/support/KafkaRecordProcessorFacade.java:135]: this part of the code is one of 2 reasons I can see that we need the offset in `ProcessedResult`
may be worth trying to move it inside `createExchange`

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java [hunk 3]
```diff
@@ -297,6 +297,7 @@ private void subscribe() {
     }
 
     protected void startPolling() {
+        // not sure what we are using this for
         long partitionLastOffset = -1;
```
Comment (CodeSmell) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java:301]: saw this and it is used in Polling Strategy but I didn't see where we changed it

**Comment by orpiske** [comment]
Attaching test logs from Github. 

[components-camel-kafka.log](https://github.com/apache/camel/files/13317865/components-camel-kafka.log)

**Comment by CodeSmell** [comment]
Will add another PR with this for consideration

**Pull request closed without merging**